atl-checker/src/algorithms/global/mod.rs

```diff
@@ -0,0 +1,466 @@
+use crate::edg::{Edge, ExtendedDependencyGraph, HyperEdge, NegationEdge, Vertex};
+use std::cmp::max;
+use std::collections::{HashMap, HashSet, VecDeque};
+use std::fmt::Debug;
+
+// Based on the global algorithm described in "Extended Dependency Graphs and Efficient Distributed Fixed-Point Computation" by A.E. Dalsgaard et al., 2017
+pub struct GlobalAlgorithm<
+    G: ExtendedDependencyGraph<V> + Send + Sync + Clone + Debug + 'static,
+    V: Vertex + Send + Sync + 'static,
+> {
+    edg: G,
+    /// The root vertex, which we discover the graph from
+    v0: V,
+    /// The assignment for all vertices, we use this to keep track  of the current assignment
+    assignment: HashMap<V, bool>,
+    /// The distance of a vertex is represented as the index number of which set the vertex is located in
+    dist: VecDeque<HashSet<V>>,
+}
+
+impl<
+        G: ExtendedDependencyGraph<V> + Send + Sync + Clone + Debug + 'static,
+        V: Vertex + Send + Sync + 'static,
```

Comment:
I believe `Send + Sync + Clone + Debug + 'static` are all unnecessary for the global algorithm since it is not distributed.